src/alertmanager_client.py

```diff
@@ -33,21 +34,42 @@ def reload(self) -> bool:
           True if reload succeeded (returned 200 OK); False otherwise.
         """
         url = urllib.parse.urljoin(self.base_url, "/-/reload")
-        return bool(self._get(url, timeout=self.timeout))
+        if resp := self._post(url, timeout=self.timeout):
+            logger.warning("reload: POST returned a non-empty response: %s", resp)
+            return False
+        return True
```

Comment:
If for a little moment Alertmanager API does not respond you'll get an error.
In `grafana-agent-operator` I implemented something [like this](https://github.com/canonical/grafana-agent-operator/blob/devel/src/charm.py#L314):

```python
    def _reload_config(self, attempts: int = 10) -> None:
        """Reload the config file.
        Args:
            attempts: number of attempts to reload
        """
        try:
            self.unit.status = MaintenanceStatus("reloading agent configuration")
            url = "http://localhost/-/reload"
            errors = list(range(400, 452)) + list(range(500, 513))
            s = Session()
            retries = Retry(total=attempts, backoff_factor=0.1, status_forcelist=errors)
            s.mount("http://", HTTPAdapter(max_retries=retries))
            s.post(url)
        except Exception as e:
            message = f"could not reload configuration: {str(e)}"
            raise GrafanaAgentReloadError(message)
```